log.h

```diff
@@ -30,6 +30,11 @@
 #include <string>
 #include <vector>
 #include <mutex>
+#ifdef _WIN32
+// Include mingw-std-threads extra header
+#include <mingw.mutex.h>
```

Comment:
Why is this gated on _WIN32 and not __MINGW32__ though? This would break, e.g., an MSVC build, or a Clang build.